src/cn-batch-forms.service.js

```diff
@@ -98,26 +98,10 @@ export function setValue(ffService) {
         update.set(updateVal);
       }
       else if (_.isObject(originalVal) && _.isObject(val)) {
-        let newVal = _.cloneDeep(originalVal)
-        for (let key in val) {
-          if (key in originalVal && _.isArray(val[key]) && _.isArray(originalVal[key])) {
-            let found = false;
-            for (let arrayVal of val[key]) {
-              found = false;
-              for (let ogArrayVal of originalVal[key]) {
-                if (_.isEqual(ogArrayVal, arrayVal)) {
-                  found = true;
-                  break;
-                }
-                if (!found) {
-                  newVal[key].push(arrayVal);
-                }
-              }
-            }
-          } else {
-            newVal[key] = val[key];
-          }
-        }
+        let newVal = _.merge(_.cloneDeep(originalVal), _.cloneDeep(val), (a, b) =>
+           _.isArray(a) ? [...new Set(a.concat(b))] : a
```

Comment:
Shouldn't this work as:
```suggestion
           _.isArray(a) ? new Set(a.concat(b)) : a
```

Reply:
Oh nevermind, a `Set` isn't the same as an array.

Reply:
There is `_.uniq`, so it could be:
```suggestion
           _.isArray(a) ? _.uniq(a.concat(b)) : a
```
What you have is fine though lol, I can get carried away with these things.

Reply:
Uniq seems to work as well. Thanks!

> What you have is fine though lol, I can get carried away with these things.
Love it, keep them going haha